packages/ui-grid/src/GridCol/index.tsx

```diff
@@ -67,12 +68,36 @@ class GridCol extends Component<GridColProps> {
     }
   }
 
+  widthCheck() {
+    const { width } = this.props
+    let shouldWarn = false
+    if (width) {
+      if (typeof width === 'number' && width <= 0) {
+        shouldWarn = true
+      }
+
+      if (typeof width === 'object') {
+        Object.keys(width).forEach((breakpoint) => {
+          //@ts-expect-error Ts doesn't understand Object.keys properly
+          if (typeof width[breakpoint] === 'number' && width[breakpoint] <= 0) {
+            shouldWarn = true
+          }
+        })
+      }
+    }
+    if (shouldWarn) {
+      warn(false, 'Col width must be positive!')
+    }
+  }
+
```

Comment:
Just an idea:
maybe we should warn the user if the width is over 12 as well

what do you think?